kolibri_explore_plugin/assets/src/searchApi.js

```diff
@@ -0,0 +1,56 @@
+import { KolibriApi } from './kolibriApi';
+
+function zimSearch(zimfile, query, node, maxResults = 10, start) {
+  const data = {
+    query,
+    max_results: maxResults,
+  };
+  if (start) {
+    data.start = start;
+  }
+  const { origin } = window.location;
+  const url = new URL(`${origin}/zim/search/${zimfile.checksum}.zim`);
+  for (const k in data) {
+    url.searchParams.append(k, data[k]);
+  }
+  return fetch(url)
+    .then(r => r.json())
+    .then(r => ({ ...r, node }));
+}
+
+function zimToNode(zimArticle, node) {
+  return {
+    ...zimArticle,
+    node: node,
+    id: `${node.id}${zimArticle.path}`,
+    nodeId: node.id,
+    channelId: node.channel_id,
+    description: zimArticle.snippet,
+    kind: 'zim',
+    directContent: true,
+    nodeUrl: `/topics/${node.channel_id}/c/${node.id}?zimPath=${zimArticle.path}`,
+  };
+}
+
+export function wikiChannelSearch(channelId, query) {
+  const kolibri = new KolibriApi(channelId);
+  // Get all the zim nodes in the channel
+  return kolibri
+    .getContentByFilter({
+      kinds: ['zim'],
+    })
+    .then(({ results }) => {
+      // Do the search for each zim node
+      const promises = results.map(node => {
+        const [zimfile] = node.files.filter(f => f.extension === 'zim');
+        return zimSearch(zimfile, query, node);
+      });
+      return Promise.all(promises);
+    })
+    .then(nodes => {
+      // Convert the result to a ContentNode like object
+      return nodes
+        .map(n => n.articles.map(a => zimToNode(a, n.node)))
```

Comment:
This conversion will be expensive. I wonder if we can do it when clicking on the card instead of when fetching the results.